### PR TITLE
Add offered qos incompatible event and event executor test

### PR DESCRIPTION
**Pull request by ivanpauno**
(edit) #10 has already been merged.

This PR is build on top of https://github.com/osrf/ros2_java/pull/10.

It adds a new publisher event status: `OfferedQosIncompatible`.
That new event status is used to test the executor code in https://github.com/osrf/ros2_java/pull/10, which surprisingly works without changes.

There are some other cosmetic changes here.

I will rebase and redirect the PR to `feature/events` when #10 gets merged.

### CODE REVIEW

File: rcljava/src/main/java/org/ros2/rcljava/events/publisher_statuses/OfferedQosIncompatible.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  static {
+    try {
+      JNIUtils.loadImplementation(OfferedQosIncompatible.class);
+    } catch (UnsatisfiedLinkError ule) {
+      logger.error("Native code library failed to load.\n" + ule);
+      System.exit(1);
+    }
+  }
+
+  private static native long nAllocateRCLStatusEvent();
```
Comment (jacobperron) [inline rcljava/src/main/java/org/ros2/rcljava/events/publisher_statuses/OfferedQosIncompatible.java:72]: nit: why change conventions and shorten the prefix from `native` to `n`? I'd prefer we be consistent and spell-out `native` for now.

File: rcljava/src/test/java/org/ros2/rcljava/SpinTest.java [hunk 2]
```diff
@@ -333,4 +343,62 @@ public Node getNode() {
...
+  }
+
+  @Test
+  public final void testSpinEvent() {
+    String identifier = RCLJava.getRMWIdentifier();
+    if (identifier == "rmw_fastrtps_cpp" || identifier == "rmw_fastrtps_dynamic_cpp") {
+      // OfferedQosIncompatible event not supported in these implementations
+      return;
+    }
+    System.out.print(identifier);
```
Comment (jacobperron) [inline rcljava/src/test/java/org/ros2/rcljava/SpinTest.java]: I think we can remove this print statement.

**Comment by jacobperron** [comment]
> surprisingly works without changes.

:joy:

**Review by jacobperron — APPROVED** [review]
Besides the minor comments above, this LGTM

File: rcljava/src/main/java/org/ros2/rcljava/events/publisher_statuses/OfferedQosIncompatible.java (hunk 1 above)
Comment (ivanpauno) [reply rcljava/src/main/java/org/ros2/rcljava/events/publisher_statuses/OfferedQosIncompatible.java:72]: The issue was that some of the mangled C symbols were too long, and I didn't find a way to silent the linter correctly.

I'm planning to rename the package as `org/ros2/rcljava/events/statuses/pub`, as underscores aren't an usual practice in java package names. Maybe that solves the line length problem without changing `native` to `n`.

File: rcljava/src/test/java/org/ros2/rcljava/SpinTest.java (hunk 2 above)
Comment (ivanpauno) [reply rcljava/src/test/java/org/ros2/rcljava/SpinTest.java]: Thanks! That was a leftover from debugging.
Solved in https://github.com/osrf/ros2_java/pull/11/commits/99517698337da063ff5d5bec36ac5f1f01d227de.

**Comment by ivanpauno** [comment]
I have rebased `feature/events` against `foxy-devel`, and this branch against `feature/events`.
In that way, we can see CI being exercised now :smiley: .

File: rcljava/src/test/java/org/ros2/rcljava/SpinTest.java (hunk 2 above)
Comment (ivanpauno) [reply rcljava/src/test/java/org/ros2/rcljava/SpinTest.java]: I forgot I added that print for debugging, but I never fixed the issue :joy:.
Fixed in https://github.com/osrf/ros2_java/pull/11/commits/ce1ba9f1708c2f35713e19892f77977049f6ccfe.

**Comment by ivanpauno** [comment]
@jacobperron there are some event statuses that we want but are not available in foxy (e.g. message lost), should we create a `galactic-devel` branch?

(edit) that should be done after completing basic event handling support in the foxy branch

**Comment by ivanpauno** [comment]
Unrelated, but I want to get rid of all [stale branches](https://github.com/osrf/ros2_java/branches/stale).
Do we need to keep some of them or can I delete them all?

File: rcljava/src/main/java/org/ros2/rcljava/events/publisher_statuses/OfferedQosIncompatible.java (hunk 1 above)
Comment (ivanpauno) [reply rcljava/src/main/java/org/ros2/rcljava/events/publisher_statuses/OfferedQosIncompatible.java:72]: We can solve this in a follow up, let me know if you have any ideas.

Comment (ivanpauno) [reply rcljava/src/main/java/org/ros2/rcljava/events/publisher_statuses/OfferedQosIncompatible.java:72]: With the changes in https://github.com/osrf/ros2_java/pull/12 and https://github.com/osrf/ros2_java/pull/14 the problem gets solved.

**Comment by jacobperron** [comment]
@ivanpauno 

>  there are some event statuses that we want but are not available in foxy (e.g. message lost), should we create a galactic-devel branch?

Yeah, feel free to create a `galatic-devel` branch when the time is right.

> Unrelated, but I want to get rid of all stale branches.

Go ahead and delete stale branches :+1:

**Pull request merged**